### ISSUE TITLE
[REQUEST] Migrate tallclair/mdtoc to k-sigs/mdtoc

### ISSUE THREAD

**Issue by saschagrunert** [kubernetes/org#1875]
### New Repo, Staging Repo, or migrate existing
Both would be fine, either we migrate https://github.com/tallclair/mdtoc or create a new repository and move the source code over there.

### Requested name for new repository
mdtoc

### Which Organization should it reside
kubernetes-sigs

### If not a staging repo, who should have admin access
I guess (correct me If I'm wrong)
@kubernetes/test-infra-admins 
@kubernetes/release-managers 

### If not a staging repo, who should have write access
The admins

### If not a staging repo, who should be listed as approvers in OWNERS
@tallclair 
@kubernetes/enhancements-admins 

### If not a staging repo, who should be listed in SECURITY_CONTACTS
TBD

### What should the repo description be
Markdown Table of Contents generator for golang based projects

### What SIG and subproject does this fall under in sigs.yaml
sig-release

### Approvals
Kindly requesting approval from @tallclair and @justaugustus, I'm not sure if we need other approvals here as well.

### Additional context for request
Further context can be found in: https://github.com/kubernetes/release/issues/1010

**Comment by justaugustus** [comment]
@saschagrunert -- I think @tallclair + @kubernetes/enhancements-admins should be sufficient owners to start.

I'm fine with this moving forward, but hold for Tim's explicit approval.

**Comment by mrbobbytables** [comment]
/assign @tallclair 
for approval 👍

**Comment by nikhita** [comment]
/assign

Donated repos need to follow some [extra steps](https://github.com/kubernetes/community/blob/master/github-management/kubernetes-repositories.md#rules-for-donated-repositories) before migration. Specifically, we'll need the following updates:

1. The contributing guidelines need removal of google specific instructions and should follow https://github.com/kubernetes/kubernetes-template-project/blob/master/CONTRIBUTING.md

2. `code-of-conduct.md`, `OWNERS`, `SECURITY_CONTACTS` files need to be added. Template is available here - https://github.com/kubernetes/kubernetes-template-project

3. Copyright header needs to be updated from the Google one to the Kubernetes one - https://github.com/kubernetes/kubernetes/tree/master/hack/boilerplate.

1-2 can be updated by a PR from anyone, but 3 needs to be updated by @tallclair or a Google representative for legal reasons.

**Comment by saschagrunert** [comment]
> 1-2 can be updated by a PR from anyone, but 3 needs to be updated by @tallclair or a Google representative for legal reasons.

Ready for review there https://github.com/tallclair/mdtoc/pull/1

**Comment by tallclair** [comment]
Sorry, for some reason I thought I had licensed this personally, but since Google owns the copyright I'll need to get approval from our opensource office first. I've started that conversation, and will report back when the transfer is unblocked. I don't foresee any problems.

**Comment by nikhita** [comment]
@tallclair :wave: Just following up. Have there be any updates on the licensing front?

**Comment by tallclair** [comment]
Yes, making some progress. I think it's almost done.

**Comment by tallclair** [comment]
Approved! We're good to move forward.

**Comment by saschagrunert** [comment]
Thank you @tallclair! 🙏 @nikhita how does the repo move work technically?

**Comment by nikhita** [comment]
That's awesome! The copyright header still needs to be updated from the Google one to the k8s one (https://github.com/kubernetes/org/issues/1875#issuecomment-629178949), but we'd be good to go after that. :+1: 

> @nikhita how does the repo move work technically?

GitHub allows repo transfers only between orgs i.e. the repo needs to be in an org instead of a personal account. To move it to k-sigs, we'll need to move it to a dummy org and then move it from there to k-sigs. I'll reach out to @tallclair about this once the copyright headers have been updated. :)

**Comment by tallclair** [comment]
Copyright updated! https://github.com/tallclair/mdtoc/pull/2 (sorry it took me so long 🙄 )

How do we do the transfer?

**Comment by nikhita** [comment]
Repo has been migrated -- https://github.com/kubernetes-sigs/mdtoc

@saschagrunert -- want to double check, does everyone in @kubernetes/test-infra-admins and @kubernetes/release-managers need admin access to this repo? That's a total of 18 members. Ideally, we prefer to keep admin access scoped to only a few folks -- usually it's the repo OWNERS.

**Comment by tallclair** [comment]
I'd like to add CI to the repo. Do we have a standard CI setup for kubernetes subprojects that can easily be enabled on this repo, or should I just add the GitHub golang action (just build & test)?

**Comment by nikhita** [comment]
> Do we have a standard CI setup for kubernetes subprojects that can easily be enabled on this repo

You can use prow :smile: 

https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#how-to-configure-new-jobs has detailed instructions on how to write a prowjob.

TLDR:

- You'll need to create a directory for `mdtoc` here - https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes-sigs
- Add a `mdtoc-presubmits.yaml` file with separate jobs for build/test similar to https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/release-notes/release-notes-presubmits.yaml

**Comment by tallclair** [comment]
Thanks. Filed https://github.com/kubernetes-sigs/mdtoc/issues/3 for now, but I'll try to get to it later this week.

**Comment by saschagrunert** [comment]
> @saschagrunert -- want to double check, does everyone in @kubernetes/test-infra-admins and @kubernetes/release-managers need admin access to this repo? That's a total of 18 members. Ideally, we prefer to keep admin access scoped to only a few folks -- usually it's the repo OWNERS.

I think the release manager do not need admin access to the repo, but the `sig-release-leads`. Since the project would be only used in k/enhancements and k/release for now I think that would be the only group to add.

**Comment by saschagrunert** [comment]
Proposed the change here: https://github.com/kubernetes-sigs/mdtoc/pull/5
